optimizer/resolver.go

```diff
@@ -429,7 +467,12 @@ func (nr *nameResolver) resolveColumnInResultFields(cn *ast.ColumnNameExpr, rfs
 	var matched *ast.ResultField
 	for _, rf := range rfs {
 		matchAsName := cn.Name.Name.L == rf.ColumnAsName.L
-		matchColumnName := rf.ColumnAsName.L == "" && cn.Name.Name.L == rf.Column.Name.L
+		var matchColumnName bool
+		if ctx.checkOriginName {
+			matchColumnName = cn.Name.Name.L == rf.Column.Name.L
+		} else {
+			matchColumnName = rf.ColumnAsName.L == "" && cn.Name.Name.L == rf.Column.Name.L
```

Comment:
we can just remove `rf.ColumnAsName.L == ""`, so we don't need  `checkOriginName`.